ldb/block.go

```diff
@@ -290,3 +290,41 @@ func (db *LevelDb) NewestSha() (rsha *btcwire.ShaHash, rblkid int64, err error)
 
 	return &sha, db.lastBlkIdx, nil
 }
+
+// fetchAddrIndexTip returns the last block height and block sha to be indexed.
+// Meta-data about the address tip is currently cached in memory, and will be
+// updated accordingly by functions that modify the state. This function is
+// used on start up to load the info into memory. Callers will use the public
+// version of this function below, which returns our cached copy.
+func (db *LevelDb) fetchAddrIndexTip() (*btcwire.ShaHash, int64, error) {
+	db.dbLock.Lock()
+	defer db.dbLock.Unlock()
+
+	data, err := db.lDb.Get(addrIndexMetaDataKey, db.ro)
+	if err != nil {
+		return &btcwire.ShaHash{}, -1, btcdb.ErrAddrIndexDoesNotExist
```

Comment:
This answers my question above.. but why not return nil here instead? Otherwise we end up creating additional garbage for no good reason, for a value that shouldn't be touched to begin with.